vividus-facade-jira/src/main/java/org/vividus/jira/JiraClientProvider.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.jira;
+
+import static org.apache.commons.lang3.Validate.isTrue;
+
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpStatus;
+import org.vividus.http.client.HttpClientConfig;
+import org.vividus.http.client.IHttpClient;
+import org.vividus.http.client.IHttpClientFactory;
+import org.vividus.http.handler.HttpResponseHandler;
+import org.vividus.http.handler.StatusCodeValidatingHandler;
+import org.vividus.jira.model.JiraConfiguration;
+import org.vividus.util.property.PropertyMappedCollection;
+
+public class JiraClientProvider
+{
+    private final PropertyMappedCollection<JiraConfiguration> jiraConfigurations;
+    private final IHttpClientFactory httpClientFactory;
+
+    private final LoadingCache<JiraConfiguration, JiraClient> jiraClients = CacheBuilder.newBuilder()
+            .build(new CacheLoader<>()
+            {
+                @Override
+                public JiraClient load(JiraConfiguration configuration) throws Exception
+                {
+                    HttpClientConfig clientConfig = configuration.getHttpClientConfig();
+                    @SuppressWarnings("MagicNumber")
+                    HttpResponseHandler statusCodeHandler = new StatusCodeValidatingHandler(HttpStatus.SC_OK, 299,
```

Comment:
is it really required to have this much-relaxed range?

Reply:
By http rfc the success range is from 200 up to 299 https://datatracker.ietf.org/doc/html/rfc2616#section-10.2

Reply:
Yep, but 227+ is unassigned :)